cmake/BLTOptions.cmake

```diff
@@ -43,7 +43,8 @@ option(ENABLE_CLANG_CUDA   "Enable Clang's native CUDA support" OFF)
 mark_as_advanced(ENABLE_CLANG_CUDA)
 set(BLT_CLANG_CUDA_ARCH "sm_30" CACHE STRING "Compute architecture to use when generating CUDA code with Clang")
 mark_as_advanced(BLT_CLANG_CUDA_ARCH)
-option(ENABLE_ROCM         "Enable ROCM support" OFF)
+option(ENABLE_HIP         "Enable HIP support" OFF)
+option(ENABLE_HCC         "Enable HCC support" OFF)
 set(BLT_ROCM_ARCH "gfx900" CACHE STRING "gfx architecture to use when generating ROCm code")
```

Comment:
Let me try and answer your second point. AMD has a c++amp-like programming model for targeting our GPUs called HC, and using the hcc compiler. AMD has also developed the HIP API and hipcc compiler, which serves as a thin layer between an application and CUDA or HC. The hipcc compiler detects (or is told) what architecture to compile for and either inlines CUDA calls or equivalent HC functions at compile time. HIP's syntax is almost identical to CUDA's. While this sounds like just a thin portability layer, it has actually become the preferred development approach for folks using or transitioning to AMD's GPU. This could be due to projects already having existing CUDA code, or just developers being familiar with the CUDA programming model in general.

As for your first question, both the hcc and hipcc compiler can take an architecture argument to specify what device code to generate. I can call BLT_ROCM_ARCH whatever you think makes sense style-wise. The switch in general from using ROCM to HCC in some existing parts of BLT can be discussed too, as it's also just a style choice and can be changed/reverted.

Reply:
Thanks @noelchalmers -- that's really helpful.

Based on your description (where you don't mention ROCM in the first paragraph), perhaps blt should only refer to hip and hcc and avoid referring to rocm.  More terms might simplify things for our users. 

What do others think?

Reply:
I'm happy with `BLT_ROCM_ARCH` since ROCM is the underlying stack that HCC and HIP are using.

Reply:
Thanks @noelchalmers and @davidbeckingsale